typing/includemod.ml

```diff
@@ -550,16 +551,27 @@ and try_modtypes ~in_eq ~loc env ~mark subst mty1 mty2 orig_shape =
     end
   | (Mty_signature sig1, Mty_signature sig2) ->
       begin match
-        signatures ~in_eq ~loc env ~mark subst sig1 sig2 orig_shape
+        signatures ~in_eq ~loc env ~mark subst ~modes sig1 sig2 orig_shape
       with
       | Ok _ as ok -> ok
       | Error e -> Error (Error.Signature e)
       end
 
   | Mty_functor (param1, res1), Mty_functor (param2, res2) ->
+      let modes_param, modes_res =
+        match modes with
+        | None ->
+            (* comparing two modules type decls, in which case we compare functor
+            parameter and return as module type decls. *)
+            None, None
```

Comment:
This actually doesn't look quite right to me. I don't think whether we are checking an inclusion as part of a `module type` check affects what has to happen here. For example, I think that this check should pass:
```ocaml
(struct
  module type S = functor (X : sig val x : string @@ global end) -> sig end
  end: sig
   module type S = functor (X : sig val x : string end) -> sig end
 end)
```
the two `S`s are mutually convertible without runtime coercion at any mode -- which is what the definition of module type equality is.

This is different from:
```ocaml
(struct
  module type S = sig val x : string @@ global
  end: sig
   module type S = sig val x : string
 end)
 ```
 because here the two `S`s are not mutually convertible at all modes: they would not be convertible at `local`. Of course we currently don't support `local` modules, but we don't want to make our type equality bake in that fact as we'd like to change it later.
 
(I think I got myself confused in a discussion with you the other day around this area, and strongly suspect that is what lead you to write it this way. Sorry about that).

Reply:
As discussed offline, we agree this is the right way to go. Fixed in commit "correct semantics".